src/runner.py

```diff
@@ -149,6 +163,10 @@ class RunnerSingleJob(Runner):
     def _run_single_job(self, checkout_node, plan, device):
         try:
             job, tmp = self._schedule_test(checkout_node, plan, device)
+            if not job and not tmp:
```

Comment:
Isn't it enough to have just `job` or `tmp` having a boolean value of `False` to mark job as failed?

```suggestion
            if not job or not tmp:
```

or maybe it's just enough to check for `job` being evaluated to False?

```suggestion
            if not job:
```

Reply:
btw is `tmp`  used anywhere in this method? If not it should rather by renamed to `_`

```python
    job, _ = self._schedule_test(checkout_node, plan, device)
```

Reply:
`tmp` is useful to display the error message.
Updated the code accordingly.

Reply:
Why not have the whole logic in to the main try-except block rather than adding code especially for handling this?

Reply:
`RunnerLoop.loop` already has try-except block implemented which is causing the runner to be terminated after catching exceptions.
API can raise exceptions while creating nodes. Hence, added a special block in `create_node`.
Seems like I would need to look into this again as service base class implementation just got merged.

Reply:
I guess we need to be careful if we want to keep the runner running even when the API is throwing exceptions.  It's probably safer to let it terminate for now, and see if we need to add some "robustness" later on without compromising anything.

Reply:
For POST node requests, there is a check to validate the parent (if the parent is in an available state) before creating child nodes in the API. This can result in an exception if the parent is invalid.
I have added exception handling specifically for `create_node` to handle this case.
I don't think the `runner` should be terminated if somehow the test node field for a specific revision is invalid.
As this can cause prevention of running tests (maybe with valid data) for other revisions.

Reply:
The call to the API is returning `HTTPException` with `400` status if the parent node is not in the available state.  I guess the runner shouldn't crash in a situation when this happens, but it'd be better to narrow down the error handling to `HTTPError` instead of silencing all possible errors with the generic `except Exception` clause.

Reply:
I agree with this.
Fixed the issue.